SSA/Experimental/IntrinsicAsymptotics.lean

```diff
@@ -15,10 +15,23 @@ open Goedel (toType)
   # Classes
 -/
 
+structure Signature (Ty : Type) where
+  sig     : List Ty
+  regSig  : List (Ctxt Ty × Ty)
+  outTy   : Ty
+
 class OpSignature (Op : Type) (Ty : outParam (Type)) where
-  sig : Op → List Ty
-  regSig : Op → List (Ctxt Ty × Ty)
-  outTy : Op → Ty
+  signature : Op → Signature Ty
+export OpSignature (signature)
+
+section
+variable {Op Ty} [s : OpSignature Op Ty]
+
+def OpSignature.sig     := Signature.sig ∘ s.signature
+def OpSignature.regSig  := Signature.regSig ∘ s.signature
+def OpSignature.outTy   := Signature.outTy ∘ s.signature
```

Comment:
```suggestion
@[match_pattern]
abbrev OpSignature.sig     := Signature.sig ∘ s.signature
@[match_pattern]
abbrev OpSignature.regSig  := Signature.regSig ∘ s.signature
@[match_pattern]
abbrev OpSignature.outTy   := Signature.outTy ∘ s.signature
```

this allows matching as `.sig` and such, and also should allow better unfolding in the rare case that these show in in proofs?

Reply:
Let's dot this once the need arises.